packages/flutter/lib/src/rendering/sliver_fill.dart

```diff
@@ -124,7 +134,7 @@ class RenderSliverFillRemaining extends RenderSliverSingleBoxAdapter {
     assert(paintedChildSize.isFinite);
     assert(paintedChildSize >= 0.0);
     geometry = SliverGeometry(
-      scrollExtent: constraints.viewportMainAxisExtent,
+      scrollExtent: scrollBody ? constraints.viewportMainAxisExtent : constraints.scrollOffset,
```

Comment:
Setting the scrollExtend to the scrollOffset feels odd. Why does the first visible pixel of this sliver (scrollOffset) determine how much content this sliver has to provide (scrollExtend)?

Reply:
`scrollOffset` can refer to the beginning or end of the sliver's contents.  Particularly in this case, I believe it is relevant because the top of the sliver is not past the top of the viewport, as well as it being not past the bottom of the viewport. In this use case of `SliverFillRemaining`, the sliver fits within the viewport in both directions.